### PR TITLE
categories page table init

### PR DESCRIPTION
**Pull request by biskuvit**
Test MUI finefoods admin! 'MUI'
[Link to MUI-FINE-FOOD-CATEGORIES-PAGE](https://mui-fin-mui.refine.dev)

Test me! 'MUI'
[Link to MUI-FINE-FOOD-CATEGORIES-PAGE](https://mui-fin-refine.pankod.com)

I am going to create a categories page.

- [X]  Categories editable table
- [X]  Expanded table
- [x] product edit drawer isActive type bug

Test me! 'MUI'
[Link to MUI-FINE-FOOD-CATEGORIES-PAGE](https://mui-fin-refine.pankod.com)

### CODE REVIEW

**Review by omeraplak — APPROVED** [review]

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 1]
```diff
@@ -0,0 +1,437 @@
+import React, { useCallback } from "react";
+import { useTranslate, IResourceComponentsProps } from "@pankod/refine-core";
+import { ICategory, ICourier, IProduct } from "interfaces";
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:3]: local import should relocate

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:4]:
```suggestion
```

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+import { useForm, useModalForm } from "@pankod/refine-react-hook-form";
+import { useTable, Column, useExpanded } from "@pankod/refine-react-table";
+import {
+    Edit,
+    AddCircleOutline,
+    RemoveCircleOutline,
+} from "@mui/icons-material";
+
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:12]:
```suggestion
```

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+    const t = useTranslate();
+    const columns: Array<Column> = React.useMemo(
+        () => [
+            {
+                id: "title",
+                Header: t("categories.fields.title"),
+                accessor: "title",
+                // eslint-disable-next-line react/display-name
+                Cell: ({ row }: any) => {
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:62]: could we specify the type?

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+                        </Stack>
+                    );
+                },
+            },
+            {
+                id: "isActive",
+                Header: t("categories.fields.isActive"),
+                accessor: "isActive",
+                // eslint-disable-next-line react/display-name
+                Cell: ({ value }: any) => <BooleanField value={value} />,
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:82]: the type needs to specify

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+            },
+            {
+                id: "actions",
+                Header: "Actions",
+                accessor: "id",
+                //eslint-disable-next-line react/display-name
+                Cell: ({ value }: any) => {
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:89]: the type needs to specify

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+                        10,
+                        25,
+                        { label: "All", value: pageOptions?.length },
+                    ]}
+                    showFirstButton
+                    showLastButton
+                    count={pageCount}
+                    rowsPerPage={pageSize}
+                    page={pageIndex}
+                    onPageChange={(event: unknown, newPage: number) =>
```
Comment (ozkalai) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:299]: Could we specify the event type?

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+    TablePagination,
+    useDataGrid,
+    Avatar,
+    GridColumns,
+    DataGrid,
+    DateField,
+    NumberField,
+    GridActionsCellItem,
+} from "@pankod/refine-mui";
+import { ICategory, IProduct } from "interfaces";
```
Comment (salihozdemir) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:41]:
```suggestion

import { ICategory, IProduct } from "interfaces";
```

**Review by salihozdemir — APPROVED** [review]

**Review by ozkalai — APPROVED** [review]

File: examples/fineFoods/admin/mui/src/pages/categories/list.tsx [hunk 9]
```diff
@@ -0,0 +1,450 @@
...
+    const renderRowSubComponent = useCallback(
+        ({ row }) => <CategoryProductsTable record={row.original} />,
+        [],
+    );
+
+    const handleEditButtonClick = (editId: string) => {
+        setId(editId);
+    };
+
+    const renderEditRow = useCallback(
```
Comment (omeraplak) [inline examples/fineFoods/admin/mui/src/pages/categories/list.tsx:161]: apparently there are lint errors

**Review by omeraplak — CHANGES_REQUESTED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**